templates/user/view/page/user-name.php

```diff
@@ -1,5 +1,8 @@
 <?php
 // Exit if accessed directly.
 defined( 'ABSPATH' ) || exit;
+
+$online_users = (array) get_transient( 'online_users' );
 ?>
+<span class="hp-vendor__online-status <?php echo in_array( $user->get_id(), array_keys( $online_users ) ) && $online_users[ $user->get_id() ] > time() - 60 ? 'online' : 'offline'; ?>"></span>
```

Comment:
Please use BEM notation, online/offline are modifiers so it should be:

.hp-vendor__online-status--online

But I recommend creating a common CSS class, like .hp-activity-badge (it's an element which doesn't depend on the block, we'll use it for both vendors and users, or even in message threads where blocks have different CSS classes), with modifiers:

.hp-activity-badge (for offline, no need to define it since the dot is just grey by default)
.hp-activity-badge .hp-activity-badge--online (the modifier class adds green color), in LESS it looks like:

&-activity-badge {
    // default styles

    &--online {
        // styles for online
    }
}

We have a similar class naming for the verified badge.